src/brod_group_subscriber.erl

```diff
@@ -379,8 +379,12 @@ handle_call({assign_partitions, Members, TopicPartitions}, _From,
             #state{ cb_module = CbModule
                   , cb_state  = CbState
                   } = State) ->
-  Result = CbModule:assign_partitions(Members, TopicPartitions, CbState),
-  {reply, Result, State};
+  case CbModule:assign_partitions(Members, TopicPartitions, CbState) of
+    {NewCbState, Result} ->
+      {reply, Result, State#state{ cb_state = NewCbState }};
+    Result ->
```

Comment:
please add a `when is_list(Result) ` guar for this clause.
aso add a comment line to explain why we should keep this clause.